sotodlib/preprocess/processes.py

```diff
@@ -538,6 +514,86 @@ def plot(self, aman, proc_aman, filename):
             plot_4f_2f_counts(aman, filename=filename.replace('{name}', f'{ufm}_4f_2f_counts'))
             plot_hwpss_fit_status(aman, proc_aman[self.calc_cfgs["hwpss_stats_name"]], filename=filename.replace('{name}', f'{ufm}_hwpss_stats'))
 
+    @classmethod
+    def gen_metric(cls, meta, proc_aman):
+        """ Generate a QA metric for the coefficients of the HWPSS fit.
+        Coefficient percentiles and mean are recorded for every mode and detset.
+
+        Arguments
+        ---------
+        meta : AxisManager
+            The full metadata container.
+        proc_aman : AxisManager
+            The metadata containing just the output of this process.
+
+        Returns
+        -------
+        line : dict
+            InfluxDB line entry elements to be fed to
+            `site_pipeline.monitor.Monitor.record`
+        """
+        # record one metric per wafer_slot per bandpass
+        # extract these tags for the metric
+        tag_keys = ["wafer_slot", "tel_tube", "wafer.bandpass"]
+        tags = []
+        vals = []
+        from ..qa.metrics import _get_tag, _has_tag
+        import re
+        for bp in np.unique(meta.det_info.wafer.bandpass):
+            for ws in np.unique(meta.det_info.wafer_slot):
+                subset = np.where(
+                    (meta.det_info.wafer_slot == ws) & (meta.det_info.wafer.bandpass == bp)
+                )[0]
+
+                # get the coefficients for every detector
+                coeff = proc_aman.hwpss_stats.coeffs[subset]
+                # mask those that were not set
+                nonzero = np.any(coeff != 0.0, axis=1)
+
+                # calculate amplitude of each mode
+                mode_labels = list(proc_aman.hwpss_stats.modes.vals)
+                num_re = re.compile("^[SC]([\d])$")
```

Comment:
Is this limited to a single digit number between 0-9? I think we could have >=10f.

Reply:
Indeed I was only anticipating up to one digit here. I'll change that. Good catch!